src/App.js

```diff
@@ -1,16 +1,34 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const totalLikes = messages.reduce((total, entry) => {
+    total += entry.liked;
+    return total;
+  }, 0);
+
+  const setLike = (id) => {
+    setMessages((prev) => {
+      return prev.map((entry) => {
+        if (id === entry.id) {
+          return { ...entry, liked: !entry.liked };
+        } else { return entry; }
+      });
+    });
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat Log between {messages[0].sender} & {messages[1].sender}</h1>
```

Comment:
👍 Nice dynamic header calculation using the senders from the message data. However, Consider the case where maybe the sender who starts off the chat sends multiple messages before the recipient replies. In that case, we might look for a different way to extract the names we need for the header.